### DISCUSSION TITLE
I can't get IBus STT working

### DISCUSSION THREAD

**Discussion by sirgazil** [PhilippeRo/IBus-Speech-To-Text#7] [Q&A]
Hi,

I've been using ibus for several years to type in different languages. Recently I discovered your input method and couldn't wait to try it out. However, after installing it and setting it up, I can't get it to work anywhere. So, if you could help me figure out what is missing, that would be great.

## What I've done so far

1. Install IBus and IBus STT in my OS
2. In GNOME Settings → Keyboard, add IBus STT to my list of input methods (it is displayed as `Other (Speech To Text)`)
3. In IBus STT Settings → General:

    + Preload models: ON
    + Active on start: ON
    + System locale: OFF
    + Locales: Add lightweight models for English (USA), Spanish (Colombia), Japanese

4. In IBus STT Settings → Text formatting:

    + Added custom shortcuts for English, Spanish and Japanese and the software converted my utterances in each language correctly.

5. In the desktop, start an app and place the text cursor in a text field (tried on IceCat, Ungoogled Chromium, GNOME Text Editor, GNOME Shell)
6. Switch to IBus STT input method using Super+Space (it takes about 1 second to load).
7. Speak to the microphone and wait for speech to text conversion

    + Also tried "dictation", followed by some utterance.

## Expected result

I'm not sure because I haven't seen IBus STT in action, but I would expect text to appear in the text field in step 5.

## Unexpected result

No text is added to the text field I selected in step 5.

## System information

~~~
OS: Guix System x86_64 
Kernel: 6.1.5-gnu 
DE: GNOME 42.4 (tried on Wayland and Xorg)
ibus: 1.5.27
ibus stt: 0.4.0
~~~

## Notes

1. The OS I use has known issues when using some input methods (e.g. [issue 35610](https://issues.guix.gnu.org/35610)), but sometimes these are fixed by removing caches in `~/.cache`. That doesn't work in this case though.

## Questions

1. Do you know any video where I can see IBus STT in action?
2. Do you know what could be failing in my case?

**Comment by PhilippeRo** [comment]
Hi,

could you send me the result of  running this command in a console:
/usr/libexec/ibus-engine-stt --gapplication-replace -g

Try to activate ibus-stt with a text app of your choice.
Thanks

**Reply by sirgazil** [reply]
Hi @PhilippeRo ,

Thanks for pointing me to that command. It made me realized that the problem seems to be on the [package definition for IBus STT](https://packages.guix.gnu.org/packages/ibus-speech-to-text/) in my OS.

For example, currently, I get the following error when I run that command you provided:

~~~
$ /gnu/store/HASH-ibus-speech-to-text-0.4.0/libexec/ibus-engine-stt --gapplication-replace -g
Traceback (most recent call last):
  File "/gnu/store/87dgkps8sp1jb4z2w4m1dnp0zvnq050l-ibus-speech-to-text-0.4.0/share/ibus-stt/main.py", line 38, in <module>
    from sttenginefactory import STTEngineFactory
  File "/gnu/store/87dgkps8sp1jb4z2w4m1dnp0zvnq050l-ibus-speech-to-text-0.4.0/share/ibus-stt/sttenginefactory.py", line 26, in <module>
    from sttengine import STTEngine
  File "/gnu/store/87dgkps8sp1jb4z2w4m1dnp0zvnq050l-ibus-speech-to-text-0.4.0/share/ibus-stt/sttengine.py", line 29, in <module>
    gi.require_version('Gtk', '4.0')
  File "/gnu/store/5ijhwr127smgxrp5dls7dxnr9jwa4m8p-python-pygobject-3.40.1/lib/python3.9/site-packages/gi/__init__.py", line 129, in require_version
    raise ValueError('Namespace %s not available for version %s' %
ValueError: Namespace Gtk not available for version 4.0
~~~

Which seemed to me like a common error when `python-pygobject` and `gobject-introspection` are not in the environment. So I ran the command again in an environment as follows:

~~~
$ guix shell ibus-speech-to-text gtk python-pygobject
[env]$ /gnu/store/HASH-ibus-speech-to-text-0.4.0/libexec/ibus-engine-stt --gapplication-replace -g
~~~

I saw INFO and DEBUG information. Then launched an application as you indicated, switched to STT input method, started speaking, and text began to appear on screen :)

I'll file a [bug report on Guix website](https://issues.guix.gnu.org/61236).

Sorry for the noise, and thanks for your help.
